diff --git a/compile/BUILD b/compile/BUILD
--- a/compile/BUILD
+++ b/compile/BUILD
@@ -143,15 +143,18 @@
+filegroup(
+    name = "libcxx_msan_headers",
+    srcs = [":libcxx_msan"],
+    output_group = "include",
+)
+
 pkg_files(
     name = "msan_libs_files",
     srcs = [":libcxx_msan"],
     target_compatible_with = select({
         ":gcc_build": ["@platforms//:incompatible"],
         ":macos_build": ["@platforms//:incompatible"],
         "//conditions:default": [],
     }),
-    excludes = [
-        "**/*.h",
-        "**/include/**",
-    ],
+    excludes = [":libcxx_msan_headers"],
     prefix = "msan-libs-x86_64/lib",
     strip_prefix = ".",
 )
@@ -171,15 +174,18 @@
+filegroup(
+    name = "libcxx_tsan_headers",
+    srcs = [":libcxx_tsan"],
+    output_group = "include",
+)
+
 pkg_files(
     name = "tsan_libs_files",
     target_compatible_with = select({
         ":gcc_build": ["@platforms//:incompatible"],
         ":macos_build": ["@platforms//:incompatible"],
         "//conditions:default": [],
     }),
     srcs = [":libcxx_tsan"],
-    excludes = [
-        "**/*.h",
-        "**/include/**",
-    ],
+    excludes = [":libcxx_tsan_headers"],
     prefix = "tsan-libs-x86_64/lib",
     strip_prefix = ".",
 )
